src/util/event.rs

```diff
@@ -154,6 +154,45 @@ impl<T> Drop for Event<T> {
     }
 }
 
+/// `async!` creates an Event that will be notified after the expression
+/// get executed. `$cb` is a callback placeholder that may be used in
+/// `$expr`.
+#[macro_export]
+macro_rules! async {
+    (|$cb:ident| $expr:expr) => {
+        {
+            let e1 = $crate::util::event::Event::new();
+            let e2 = e1.clone();
+            let $cb = box move |res| e2.set(res);
+            $expr;
+            e1
+        }
+    }
+}
+
+/// `await!` waits on an Event (usually created by `async!`) and returns
+/// Option<Res>.
+#[macro_export]
+macro_rules! await{
```

Comment:
_async/await_ comes from C#. See https://msdn.microsoft.com/en-us/library/mt674882.aspx
I think the asynchronous model we use is alike, so inherit the names.